src/adapter/src/subscribe.rs

```diff
@@ -316,12 +369,38 @@ impl ActiveSubscribe {
 
 /// The reason for removing an [`ActiveSubscribe`].
 #[derive(Debug, Clone)]
-pub enum SubscribeRemovalReason {
-    /// The subscribe completed successfully.
+pub enum ComputeSinkRemovalReason {
+    /// The compute sink completed successfully.
     Finished,
-    /// The subscribe was canceled due to a user request.
+    /// The compute sink was canceled due to a user request.
     Canceled,
-    /// The subscribe was forcibly terminated because an object it depended on
+    /// The compute sink was forcibly terminated because an object it depended on
     /// was dropped.
     DependencyDropped(String),
 }
+
+/// A description of an active copy to from coord's perspective.
+#[derive(Debug)]
+pub struct ActiveCopyTo {
+    /// The user of the session that created the subscribe.
+    pub user: User,
```

Comment:
They're needed because `ctx` can become `None` when `process_response` is called, but the drop code is still be interested in the `ActiveCopyTo`s connection and user ID.